src/test/java/net/kaczmarzyk/InE2eTest.java

```diff
@@ -37,104 +37,184 @@
+
+			return customerRepo.findAll(spec);
+		}
+
+		@RequestMapping(value = "/customers-ps", params = "firstNameIn")
+		@ResponseBody
+		public Object findCustomersByFirstNameUsingParamSeparator(
+				@Spec(path = "firstName", params = "firstNameIn", paramSeparator = ",", spec = In.class) Specification<Customer> spec) {
+
 			return customerRepo.findAll(spec);
 		}
-		
+
 		@RequestMapping(value = "/customers", params = "idIn")
 		@ResponseBody
 		public Object findCustomersById(
-				@Spec(path="id", params ="idIn", spec=In.class) Specification<Customer> spec) {
-			
+				@Spec(path = "id", params = "idIn", spec = In.class) Specification<Customer> spec) {
+
+			return customerRepo.findAll(spec);
+		}
+
+		@RequestMapping(value = "/customers-ps", params = "idIn")
+		@ResponseBody
+		public Object findCustomersByIdUsingParamSeparator(
+				@Spec(path = "id", params = "idIn", paramSeparator = ";", spec = In.class) Specification<Customer> spec) {
+
 			return customerRepo.findAll(spec);
 		}
-		
+
 		@RequestMapping(value = "/customers", params = "registrationDateIn")
 		@ResponseBody
 		public Object findCustomersByRegistrationDate(
-				@Spec(path="registrationDate", params = "registrationDateIn", spec=In.class) Specification<Customer> spec) {
-			
+				@Spec(path = "registrationDate", params = "registrationDateIn", spec = In.class) Specification<Customer> spec) {
+
+			return customerRepo.findAll(spec);
+		}
+
+		@RequestMapping(value = "/customers-ps", params = "registrationDateIn")
+		@ResponseBody
+		public Object findCustomersByRegistrationDateUsingParamSeparator(
+				@Spec(path = "registrationDate", params = "registrationDateIn", paramSeparator = "\\/", spec = In.class) Specification<Customer> spec) {
 			return customerRepo.findAll(spec);
 		}
-		
+
 		@RequestMapping(value = "/customers", params = "genderIn")
 		@ResponseBody
 		public Object findCustomersByGender(
-				@Spec(path="gender", params="genderIn", spec=In.class) Specification<Customer> spec) {
-			
+				@Spec(path = "gender", params = "genderIn", spec = In.class) Specification<Customer> spec) {
+
+			return customerRepo.findAll(spec);
+		}
+
+		@RequestMapping(value = "/customers-ps", params = "genderIn")
+		@ResponseBody
+		public Object findCustomersByGenderUsingParamSeparator(
+				@Spec(path = "gender", params = "genderIn", paramSeparator = "\\|", spec = In.class) Specification<Customer> spec) {
+
 			return customerRepo.findAll(spec);
 		}
 	}
-	
+
 	@Test
 	public void findsByListOfAllowedStringValues() throws Exception {
 		mockMvc.perform(get("/customers")
 				.param("firstNameIn", "Homer", "Marge")
 				.accept(MediaType.APPLICATION_JSON))
-			.andExpect(status().isOk())
-			.andExpect(jsonPath("$").isArray())
-			.andExpect(jsonPath("$[0].firstName").value("Homer"))
-			.andExpect(jsonPath("$[1].firstName").value("Marge"))
-			.andExpect(jsonPath("$[2]").doesNotExist());
+				.andExpect(status().isOk())
```

Comment:
this formmating seems werid to me, the `andExpect` is different level of invocation `()` than `accept`